packages/integration-sdk-testing/src/jest.ts

```diff
@@ -536,7 +536,24 @@ export function toMatchDirectRelationshipSchema<T extends ExplicitRelationship>(
     newRelationshipSchema.properties._type = { const: params._type };
   }
 
-  return toMatchSchema(received, newRelationshipSchema);
+  return toMatchSchema(
+    removeUndefinedProperties(received),
```

Comment:
AJV has some strange behavior in that it doesn't treat `undefined ` as `null`. 

Therefore, using the `directRelationshipSchema`, which enforces that all properties must be primitive (no arrays or objects) doesn't allow one to produce relationships with `undefined` properties:

```ts
      {
        additionalProperties: {
          anyOf: [
            { type: 'boolean' },
            { type: 'integer' },
            { type: 'null' },
            { type: 'number' },
            { type: 'string' },
          ],
        },
```